src/utils/private.ts

```diff
@@ -0,0 +1,49 @@
+import _sodium from "libsodium-wrappers";
+import { getBotConfig, getBotContext } from "../bindings";
+import { Payload } from "../types";
+
+const CONFIG_REPO = "ubiquibot-config";
+const KEY_PATH = ".github/ubiquibot-config.yml";
+const KEY_NAME = "PSK";
+const KEY_PREFIX = "HSK_";
+
+export const getPrivateKey = async (): Promise<string | undefined> => {
+  try {
+    await _sodium.ready;
+    const sodium = _sodium;
+    const context = getBotContext();
+    const {
+      sodium: { publicKey, privateKey },
+    } = getBotConfig();
+    const payload = context.payload as Payload;
+
+    if (publicKey === "" || privateKey === "") {
+      return undefined;
+    }
+
+    const { data } = await context.octokit.rest.repos.getContent({
+      owner: payload.organization?.login!,
+      repo: CONFIG_REPO,
+      path: KEY_PATH,
+      mediaType: {
+        format: "raw",
+      },
+    });
+
+    if (data) {
+      const cipherText = (data as any).startsWith(KEY_NAME) ? (data as any).replace(`${KEY_NAME}: `, "").replace(`"`, "") : undefined;
+      if (cipherText !== undefined) {
+        const binPub = sodium.from_base64(publicKey, sodium.base64_variants.URLSAFE_NO_PADDING);
+        const binPriv = sodium.from_base64(privateKey, sodium.base64_variants.URLSAFE_NO_PADDING);
+        const binCipher = sodium.from_base64(cipherText, sodium.base64_variants.URLSAFE_NO_PADDING);
+
+        let walletPrivateKey: string | undefined = sodium.crypto_box_seal_open(binCipher, binPub, binPriv, "text");
```

Comment:
So the whole idea is that a partner should keep his private key encrypted in a private repo, right?

So for a partner the flow is the following:
1. Take bot's public key
2. Encrypt partner's wallet private key with bot's public key
3. Save encrypted value in a private repo

Reply:
that's right